crates/revm/revm-inspectors/src/tracing/builder/walker.rs

```diff
@@ -0,0 +1,106 @@
+use crate::tracing::types::CallTraceNode;
+// use reth_rpc_types::trace::parity::VmTrace;
+
+/// type for doing a Depth first walk down the callgraph
+#[derive(Debug)]
+pub(crate) struct DFWalk;
+
+/// pub crate type for doing a walk down a reth callgraph
+#[derive(Debug)]
+pub(crate) struct CallTraceNodeWalker<'trace, W> {
+    /// the entire arena
+    nodes: &'trace Vec<CallTraceNode>,
+    curr_idx: usize,
+
+    /// ordered indexes of nodes
+    idxs: Vec<usize>,
+
+    phantom: std::marker::PhantomData<W>,
+}
+
+impl<'trace> CallTraceNodeWalker<'trace, DFWalk> {
+    pub(crate) fn new(nodes: &'trace Vec<CallTraceNode>) -> Self {
+        let mut idxs: Vec<usize> = Vec::with_capacity(nodes.len());
+
+        Self::get_all_children(nodes, &mut idxs, 0);
+
+        Self { nodes, curr_idx: 0, idxs, phantom: std::marker::PhantomData }
+    }
+
+    /// uses recursion to do a DFS down the callgraph
+    fn get_all_children(nodes: &'trace Vec<CallTraceNode>, holder: &mut Vec<usize>, idx: usize) {
+        holder.push(idx);
+        for child_idx in nodes[idx].children.iter() {
+            Self::get_all_children(nodes, holder, *child_idx);
+        }
```

Comment:
please no recursive calls, this can be written iteratively

Reply:
yup - context is these overflow for deep traces

Reply:
got it....

I also use recursion to crate the `VmTrace`, I should also try to change that?

Reply:
as well as for filling the code fields later on

but for now will work on making this iterative :)